packages/block-editor/src/components/inserter/utils.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Normalizes an inserter block types list and includes variations as separate items.
+ *
+ * @param {Array} items Denormalized inserter items
+ * @return {Array} Normalized inserter items.
+ */
+export function includeVariationsInInserterItems( items ) {
```

Comment:
I think should probably be bundled in the `getInserterItems` selector. But I'd leave that for a separate PR as this is already how it works today in the global inserter, it's just extracted as a function. cc @gziolo @noisysocks

Reply:
I need to double-check the initial implementation for block variation tomorrow. I think there was a reason to keep it as is initially. It might be no longer valid concern :)

Reply:
I found related PR: #19243. I can't think of any blocker for further refactoring. It seems like producing multiple items for blocks with variations followed-up with array flattening would do the trick. We just need to give it a try :)